src/BassetManager.php

```diff
@@ -464,4 +464,96 @@ public function fetchContent(string $url): string
             ->get($url)
             ->body();
     }
+
+    private function replaceAsset(CacheEntry $asset, CacheEntry $mapped, $output): StatusEnum
+    {
+        $this->disk->delete($mapped->getAssetDiskPath());
+
+        $this->cacheMap->delete($mapped);
+
+        $content = $this->getAssetContent($asset, $output);
+
+        if (! is_string($content)) {
+            return $content;
+        }
+
+        return $this->uploadAssetToDisk($asset, $content, $output);
+    }
+
+    private function getAssetContent(CacheEntry $asset, bool $output = true): StatusEnum|string
+    {
+        if (Str::isUrl($asset->getAssetPath())) {
+            // when in dev mode, cdn should be rendered
+            if ($this->dev && ! $this->forceUrlCache) {
+                $output && $this->output->write($asset, $this->dev);
+
+                return $this->loader->finish(StatusEnum::DISABLED);
+            }
```

Comment:
I see too many places in the code where dev_mode is mentioned... and I don't understand how it works any more. Previously, dev_mode meant assets weren't _actually_ internalized. Now... what does it mean?

Please explain to me:
- what `dev_mode` does in this current version;
- why you should use it;
- why you would need to separately set `dev_mode` and `force_url_cache`;

Reply:
Dev mode has 3 tasks:
1 - do not cache external asset urls
2 - do not cache files located in accessible folders (eg: /public/my-assets/asset.js)
2 - refresh the "forced" local cached files. (files that are located in the `vendor/` folder that need to be moved to accessible folders before used). 

Using the `force_url_cache` will just change the behavior of the **1** task. 

In addition to this change, dev mode will now also compute and compare the basset file hashes of the vendor files and only perform the internalization of the ones that changed.